### ISSUE TITLE
Allow redirects on cross-origin credentialed requests

### ISSUE THREAD

**Issue by taralx** [whatwg/fetch#1235]
Here's the situation I'm in:

1. There is a CORS-enabled API.
2. The API uses SSO for auth.
3. Cross-origin requests are allowed to use ambient authority (SSO).

So what happens is that when a request comes in it can be redirected to the SSO provider to get a token. That's fine, but the SSO provider does not set CORS headers, so a `fetch(..., {credentials: 'include'})` request fails. Right now this is what I'm having to do:

```
r = await fetch(api, {credentials: 'include', redirect: 'manual'})
if (r.type === 'opaqueredirect') {
  await fetch(api, {credentials: 'include', mode: 'no-cors'})
  r = await fetch(api, {credentials: 'include', redirect: 'manual'})
}
```

It's not great, since the browser ends up re-doing requests. Is there a better way to do this? Can there be one please?

**Comment by annevk** [comment]
Get the SSO provider to set the headers.

**Comment by taralx** [comment]
The SSO provider has no reason to allow cross-origin requests to access the data it returns. And I'm not even asking for that data. The rule on credentialed redirects doesn't actually make any sense to me -- if the redirect chain ends in a CORS-eligible response, why does it matter that it went through a redirect to get there?

**Comment by taralx** [comment]
At minimum if the final redirect URL matches the original request URL, the response should be eligible.

**Comment by annevk** [comment]
No, that would allow confused deputy attacks. Also, if the SSO provider does a redirect, there is no data in what it returns, right?

**Comment by taralx** [comment]
How does a URL match leave the door open to confused deputy?

**Comment by annevk** [comment]
Well, we don't know the final URL ahead of time, but there's a lot of other variables in play as well, such as method, headers, and body.

**Comment by taralx** [comment]
But that's kind of my point. I can trigger an identical series of requests with `mode: "no-cors"`, so it should be safe to run the requests to the final point and see, right?

**Comment by annevk** [comment]
How do you know it's identical? Also, in that case we purposely hide the response.

**Comment by taralx** [comment]
Well, I have to retract that because I believe one of the `Sec-Fetch-` headers exposes the mode, so they're not identical requests. Sigh.

Still not sure why we can't follow the redirect chain to see if the final request is eligible.

I'll see if I can get the SSO provider to change their stuff. I fully anticipate a bunch of "why do we need to do this" and "this isn't our fault we don't have anything to do with CORS"...
